packages/common/src/store/cache/actions.ts

```diff
@@ -17,22 +20,64 @@ export const UNSUBSCRIBE_SUCCEEDED = 'CACHE/UNSUBSCRIBE_SUCCEEDED'
 export const REMOVE = 'CACHE/REMOVE'
 export const REMOVE_SUCCEEDED = 'CACHE/REMOVE_SUCCEEDED'
 export const SET_EXPIRED = 'CACHE/SET_EXPIRED'
-export const SET_CACHE_TYPE = 'CACHE/SET_CACHE_TYPE'
+export const SET_CACHE_CONFIG = 'CACHE/SET_CONFIG'
+
+type Entry<EntryT extends Metadata = Metadata> = {
+  id: ID
+  uid?: UID
+  metadata: EntryT
+  timestamp?: number
+}
 
 /**
  * Signals to add an entry to the cache.
- * @param {Kind} kind
- * @param {array} entries { id, uid, metadata }
- * @param {boolean} replace optionally replaces the entire entry instead of joining metadata
- * @param {boolean} persist optionally persists the cache entry to indexdb
  */
-export const add = (kind, entries, replace = false, persist = true) => ({
-  type: ADD,
-  kind,
-  entries,
-  replace,
-  persist
-})
+export const add = (
+  kind: Kind,
+  entries: Entry[],
+  // optionally replaces the entire entry instead of joining metadata
+  replace = false,
+  // optionally persists the cache entry to indexdb
+  persist = true
+) => {
+  return async (dispatch, getState) => {
+    const state = await getState()
+    const confirmCalls = getConfirmCalls(state)
+    const cache = getCache(state, { kind })
+
+    const confirmCallsInCache = pick(
+      cache.entries,
+      Object.keys(confirmCalls).map((kindId) => getIdFromKindId(kindId))
+    )
+
+    const entriesToAdd = []
+    const entriesToSubscribe = []
+    entries.forEach((entry) => {
+      // If something is confirming and in the cache, we probably don't
+      // want to replace it (unless explicit) because we would lose client
+      // state, e.g. "has_current_user_reposted"
```

Comment:
will this prevent us from doing multiple local updates to the local cache, or is it ok because those would always go through update not add?

Reply:
yeah exactly ^ (also this is migrated line-for-line by the original saga that did this)